debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -581,11 +582,49 @@ public synchronized boolean isConnected() throws SQLException {
     }
 
     public synchronized Connection connection() throws SQLException {
+        return connection(true);
+    }
+
+    public synchronized Connection connection(boolean executeOnConnect) throws SQLException {
         if (conn == null) {
             conn = factory.connect(JdbcConfiguration.adapt(config));
             if (conn == null) throw new SQLException("Unable to obtain a JDBC connection");
             // Always run the initial operations on this new connection
             if (initialOps != null) execute(initialOps);
+            final String statements = config.getString(JdbcConfiguration.ON_CONNECT_STATEMENTS);
```

Comment:
I'd suggest to extract all the parsing logic into its own method which then returns the list of statements to here.

Reply:
@gunnarmorling Done